classes/author/Collector.inc.php

```diff
@@ -0,0 +1,232 @@
+
+    /** @var array|null */
+    public $publicationIds = null;
+
+    /** @var string get authors with a family name */
+    protected $familyName = null;
+
+    /** @var string get authors with a given name */
+    protected $givenName = null;
+
+    /** @var string get authors with a specified country code */
+    protected $country = null;
+
+    /** @var string get authors with a specified affiliation */
+    protected $affiliation = null;
+
+    /** @var int */
+    public ?int $count;
+
+    /** @var int */
+    public ?int $offset;
+
+    /** @var bool|null */
+    public $includeInBrowse = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Filter by contexts
+     */
+    public function filterByContextIds(?array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    /**
+     * Filter by publications
+     */
+    public function filterByPublicationIds(?array $publicationIds): self
+    {
+        $this->publicationIds = $publicationIds;
+        return $this;
+    }
+
+    /**
+     * Filter by include in browse
+     */
+    public function filterByIncludeInBrowse(bool $includeInBrowse): self
+    {
+        $this->includeInBrowse = $includeInBrowse;
+        return $this;
+    }
+
+    /**
+     * Include orderBy columns to the collector query
+     */
+    public function orderBy(string $orderBy): self
+    {
+        $this->orderBy = $orderBy;
+        return $this;
+    }
+
+    /**
+     * Filter by the given and family name
+     *
+     *
+     */
+    public function filterByName(string $givenName, string $familyName): self
+    {
+        $this->givenName = $givenName;
+        $this->familyName = $familyName;
+        return $this;
+    }
+
+    /**
+     * Filter by the specified country code
+     *
+     * @param $country string Country code (2-letter)
+     *
+     * */
+    public function filterByCountry(string $country): self
+    {
+        $this->country = $country;
+        return $this;
+    }
+
+    /**
+     * Filter by the specified affiliation code
+     *
+     * */
+    public function filterByAffiliation(string $affiliation): self
+    {
+        $this->affiliation = $affiliation;
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     */
+    public function limit(?int $count): self
+    {
+        $this->count = $count;
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     */
+    public function offset(?int $offset): self
+    {
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * @copydoc CollectorInterface::getQueryBuilder()
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $q = DB::table('authors as a')
+            ->Join('publications as p', 'a.publication_id', '=', 'p.publication_id')
```

Comment:
I think the function should be lower-case (`join`).

Reply:
changed